core-bundle/contao/library/Contao/Image.php

```diff
@@ -159,10 +159,21 @@ public static function getHtml($src, $alt='', $attributes='')
 		$search = array('{alt}', '{attributes}');
 		$replace = array(StringUtil::specialchars($alt), $attributes ? ' ' . $attributes : '');
 
+		$darkAttributes = new HtmlAttributes($attributes);
+
+		foreach (array('data-icon', 'data-icon-disabled') as $icon)
+		{
+			if (isset($darkAttributes[$icon]))
+			{
+				$pathinfo = pathinfo($darkAttributes[$icon]);
+				$darkAttributes[$icon] = $pathinfo['filename'] . '--dark.' . $pathinfo['extension'];
+			}
+		}
+
```

Comment:
This should probably be moved inside the `if (str_contains($template, '{darkAttributes}'))` condition, so the code is not run unnecessarily.

Reply:
@leofeyer Adjusted it in https://github.com/contao/contao/pull/7019/commits/15b8634934b7b3cf673021aa7b68a492ed5e93d7 and also tested that already